x-pack/plugins/rule_registry/server/rule_data_client/index.ts

```diff
@@ -102,6 +103,51 @@ export class RuleDataClient implements IRuleDataClient {
     };
   }
 
+  async rolloverAliasIfNeeded({ namespace }: { namespace?: string }) {
+    const clusterClient = await this.getClusterClient();
+    const alias = getNamespacedAlias({ alias: this.options.alias, namespace });
+    const { body: simulatedRollover } = await clusterClient.indices.rollover({
+      alias,
+      dry_run: true,
+    });
+    const writeIndexMapping = await clusterClient.indices.get({
+      index: simulatedRollover.old_index,
+    });
+    const currentVersions = get(writeIndexMapping, [
```

Comment:
why use `get` instead of the optional changing operator?

Reply:
I didn't see much difference since `currentVersions` is `any` either way. I chose this way to be consistent with how we're getting `targetVersions` below - `simulatedIndex` is typed as an empty object so the easiest way to get a field from it was just to use `get`.

Reply:
I think we can add (minimal) types for `writeIndexMapping` and `simulatedIndex` then. Once the client itself adds them we can remove it again. That way, if the types change, or are incorrect, we are notified via a TypeScript error.